server/src/main/java/org/elasticsearch/index/mapper/TsidExtractingIdFieldMapper.java

```diff
@@ -168,6 +152,32 @@ public static void createField(DocumentParserContext context, IndexRouting.Extra
         context.doc().add(new Field(NAME, uidEncoded, FIELD_TYPE));
     }
 
+    public static String createId(
```

Comment:
Only extracted the id creation from `createField()` method, 
so that the logic could also be reused in unit tests.